plugins/modules/k8s_drain.py

```diff
@@ -0,0 +1,415 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2021, Aubin Bikouo <@abikouo>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+
+module: k8s_drain
+
+short_description: Drain, Cordon, or Uncordon node in k8s cluster.
+
+author: Aubin Bikouo (@abikouo)
+
+description:
+  - Drain node in preparation for maintenance same as kubectl drain.
+  - Cordon will mark node as unschedulable.
+  - Uncordon will mark node as schedulable.
+  - The given node will be marked unschedulable to prevent new pods from arriving.
+  - Then drain deletes all pods except mirror pods (which cannot be deleted through the API server).
+
```

Comment:
```suggestion
  - Drain node in preparation for maintenance same as kubectl drain.
  - Cordon will mark the node as unschedulable.
  - Uncordon will mark the node as schedulable.
  - The given node will be marked unschedulable to prevent new pods from arriving.
  - Then drain deletes all pods except mirror pods (which cannot be deleted through the API server).

```